app/ante.go

```diff
@@ -0,0 +1,97 @@
+package app
+
+ import (
+ 	sdk "github.com/cosmos/cosmos-sdk/types"
+ 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+ 	"github.com/cosmos/cosmos-sdk/x/auth/ante"
+ 	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+ 	channelkeeper "github.com/cosmos/ibc-go/modules/core/04-channel/keeper"
+ 	ibcante "github.com/cosmos/ibc-go/modules/core/ante"
+ )
+
+ // HandlerOptions extends the SDK's AnteHandler options by requiring the IBC
+ // channel keeper.
+ type HandlerOptions struct {
+ 	ante.HandlerOptions
+
+ 	IBCChannelkeeper channelkeeper.Keeper
+ }
+
+ type MinCommissionDecorator struct{}
+
+ func NewMinCommissionDecorator() MinCommissionDecorator {
+ 	return MinCommissionDecorator{}
+ }
+
+ func (MinCommissionDecorator) AnteHandle(
+ 	ctx sdk.Context, tx sdk.Tx,
+ 	simulate bool, next sdk.AnteHandler) (newCtx sdk.Context, err error) {
+ 	msgs := tx.GetMsgs()
+ 	minCommissionRate := sdk.NewDecWithPrec(5, 2)
+ 	for _, m := range msgs {
+ 		switch msg := m.(type) {
+ 		case *stakingtypes.MsgCreateValidator:
+ 			// prevent new validators joining the set with
+ 			// commission set below 5%
+ 			c := msg.Commission
+ 			if c.Rate.LT(minCommissionRate) {
+ 				return ctx, sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "commission can't be lower than 5%")
```

Comment:
nit pick: since we use this error message elsewhere, we might want to abstract it out somewhere or even into its own error